Home assistant Basics und LDR config

sogidaned opened the pull request:
servus, 
hier ein Anfang für die Integration in Homeassistant über MQTT Funktioniert soweit auch. 
Man kann jetzt z.B. über eine Automation einen Text an der Uhr anzeigen lassen.
Die Rückmeldung vom Homeassistant zur Uhr war aber nicht ganz einfach. 
Sollte jetzt aber klappen wird dann auch ins Webinterface übertragen
![Screenshot 2025-03-10 235034](https://github.com/user-attachments/assets/23c53723-cdb7-4a1b-8c0e-db0d3151c478)


die LDR Konfiguration ist hier auch dabei. 
Die Kurve wurde Berechnet anhand meiner Konfig 4xLDR5528 und ein 10K Wiederstand
![Screenshot 2025-03-10 235209](https://github.com/user-attachments/assets/f009f0f4-8344-43f4-958c-c4c65d0a5a0f)

den Vorschlag mit dem Temperaturfühler finde ich gut. ich habe noch einen BME280 zuhause werde ich aber in einem separaten branch versuchen. 

Ich wollte es noch mit einem ESP32 testen ich bekomme aber auf meinen die ich rumliegen habe die Software nicht zum laufen.
Ich habe aber eine kleine Änderung bezüglich Auswertung vom ldr vorgenommen. da die esp32 mit 12bit auslesen stimmen sonst die eingegeben kurven nicht, da der adc wert hier bis 4095 geht

sogidaned committed:
Add files via upload

Ich habe eine Uhr mit einem bayerischen Layout. kenn mich aber nicht sonderlich gut aus hab einfach das nero Layout auf meines umgeschrieben. Lade es nur hoch falls es noch wer braucht. Kenne mich aber nicht mit Websites aus, daher nur die datei

sogidaned committed:
Merge pull request #1 from sogidaned/Bayerisches-Layout

Bayerisches Layout

sogidaned committed:
Update DE10x11.bayerisch.hpp

remove space

sogidaned committed:
add bavarian layout

sogidaned committed:
add bavarian layout

sogidaned committed:
add bavarian layout

sogidaned committed:
add bavarian layout

sogidaned committed:
add bavarian layout

sogidaned committed:
add bavarian layout

sogidaned committed:
add bavarian layout

sogidaned committed:
add bavarian layout

sogidaned committed:
"de-10-11-bayerisch": "🇩🇪 10 × 11 bavarese",

sogidaned committed:
add bavarian layout

sogidaned committed:
add bavarian layout

sogidaned committed:
Merge pull request #2 from sogidaned/Bayerisches-Layout

Bayerisches layout

sogidaned committed:
ldr auswahl und homeassistant verbesserung

sogidaned committed:
Update platformio.yml

sogidaned committed:
Update platformio.yml

sogidaned committed:
Update clang-format.yml

sogidaned committed:
Merge pull request #3 from sogidaned/ldr_auswahl

ldr auswahl und homeassistant verbesserung

sogidaned committed:
clang-format tet

sogidaned committed:
test 2

sogidaned committed:
hoffendlich letzter test

sogidaned committed:
test4

sogidaned committed:
Stufe 1 Fertig

sogidaned commented:
also die esp32 funktionieren jetzt auch lag an der zu hoch eingestellten Monitor Speed. habe sie auf 115200 runter gesetzt jetzt funktionieren beide

dbambus commented:
Hey @sogidaned, 

wollen wir mal den Pull request bis auf Fehlerverbesserungen oder Clang Anmerkungen festzurren. dann kann nicht den Mergen. :-) 

der wird aktuell ja schon ziemlich groß. 

Mit freundlichen Grüßen
David

sogidaned commented:
ich ändere hier nichts mehr dass mit der monitor Speed war nur als Hinweis gedacht. ist auch hier noch nicht geändert

dbambus commented:
Hey @sogidaned, 

ich hab mir mal den Code etwas angeschaut. Ich denke wir sollten eher nur HSB Farben verwenden, statt immer zwischen RGB und HSB umzurechnen. In der Homeassistant Discovery() wird ja in "colorMode.add("hs");" der Modus definiert und intern verwende ich nur noch HSB Farben. 

Mit freundlichen Grüßen
David

sogidaned committed:
MQTT-Implementierung optimiert: Vereinfachung und Bereinigung der Funktionen

- Entfernung redundanter MQTT-Callback-Funktionen
- Vereinheitlichung der Farbverarbeitung mit Home Assistant
- Verbesserung der Geschwindigkeitseinstellungen
- Entfernung von Debug-Code und Vereinfachung der Konfiguration

sogidaned commented:
@dbambus,

hab jetzt mal so weit es ging alles bereinigt. 
ich habe schon gesehen ich brächte einen Sensor in home assistant fürs Debugging 
da ich bei meiner Uhr keine einfache Möglichkeit habe den seriellen monitor zu nutzen da sie seit ein paar Jahren schon an der wand hängt.

timostr commented:
Gibt es eine Hochrechnung, wann Version 4.2 mit den Homeassistent Sensoren kommt?
